system/jlib/jptree.cpp

```diff
@@ -7774,20 +7766,26 @@ static void applyCommandLineOption(IPropertyTree * config, const char * option,
     }
 }
 
-static void applyCommandLineOption(IPropertyTree * config, const char * option)
+static void applyCommandLineOption(IPropertyTree * config, const char * option, std::list<std::string> ignoreOptions)
 {
     const char * eq = strchr(option, '=');
+    StringBuffer name;
+    const char *val = nullptr;
     if (eq)
     {
-        StringBuffer name;
         name.append(eq - option, option);
-        applyCommandLineOption(config, name, eq + 1);
+        option = name;
+        val = eq + 1;
     }
     else
     {
         //MORE: Support --x- and --x+?
-        applyCommandLineOption(config, option, "1");
+        val = "1";
     }
+    auto it = std::find(ignoreOptions.begin(), ignoreOptions.end(), option);
```

Comment:
style: I know it is standard c++11, but whenever I see it looks ugly.  Can we add a template function into jlib
`bool contains(T1 x, T2 search) { return (std::find(x.begin(), x.end(), search) != x.end()); }`

Reply:
yep, it is standard, but I agree also feels too long winded/verbose when I use it.
I'll add a helper.